### PR TITLE
Adding tx byte size limit for access/collection nodes

### PR DESCRIPTION
**Pull request by ramtinms**
This PR introduces a limit for the maximum byte size a transaction can take. Any transaction with more byte size will be rejected both by access node and collection nodes. The default tx byte size limit is set at 64KB.

This was planned as part of the mitigations for choking hazards. A possible side effect might be transactions trying to create several accounts or add several keys to an account. 

remaining tasks:
- [x] check emulator
- [ ] update documentation

### CODE REVIEW

**Comment by ramtinms** [comment]
@psiemens does flow emulator reuse this part of code or should I update somewhere else?

File: access/validator.go [hunk 1]
```diff
@@ -66,6 +67,11 @@ func NewTransactionValidator(
 }
 
 func (v *TransactionValidator) Validate(tx *flow.TransactionBody) (err error) {
+	err = v.checkTxSizeLimit(tx)
```
Comment (ramtinms) [inline access/validator.go:70]: I put it the first thing to check to prevent attacks on other checks.

**Comment by ramtinms** [comment]
@vishalchangrani access nodes might still need an extra byte size check for the GRPC call. this assumes we already parsed the byte array into a transaction object.

**Review by jordanschalm — APPROVED** [review]
LGTM

**Comment by jordanschalm** [comment]
> does flow emulator reuse this part of code or should I update somewhere else?

It re-uses this same validator.

https://github.com/dapperlabs/flow-emulator/blob/master/blockchain.go#L54

File: model/flow/transaction.go [hunk 2]
```diff
@@ -386,6 +406,11 @@ type ProposalKey struct {
 	SequenceNumber uint64
 }
 
+// ByteSize returns the byte size of the proposal key
+func (p ProposalKey) ByteSize() int {
+	return len(p.Address) + 8 + 8
```
Comment (vishalchangrani) [inline model/flow/transaction.go:411]:
```suggestion
       keyIDLen := 8
       sequenceNumberLen := 8
	return len(p.Address) + keyIDLen + sequenceNumberLen
```

Comment (vishalchangrani) [reply model/flow/transaction.go:411]: ☝️ just for readability

File: model/flow/transaction.go [hunk 3]
```diff
@@ -394,6 +419,11 @@ type TransactionSignature struct {
 	Signature   []byte
 }
 
+// ByteSize returns the byte size of the transaction signature
+func (s TransactionSignature) ByteSize() int {
+	return len(s.Address) + 8 + 8 + len(s.Signature)
```
Comment (vishalchangrani) [inline model/flow/transaction.go:424]:
```suggestion
        signerIndexLen := 8
        keyIDLen := 8
	return len(s.Address) + signerIndexLen + keyIDLen + len(s.Signature)
```

**Comment by vishalchangrani** [comment]
> @vishalchangrani access nodes might still need an extra byte size check for the GRPC call. this assumes we already parsed the byte array into a transaction object.

The GRPC call uses the validation you changed https://github.com/dapperlabs/flow-go/blob/master/engine/access/rpc/backend/backend_transactions.go#L48
So the AN will fail the GRPC call if the size validation fails.

**Review by vishalchangrani — APPROVED** [review]

**Pull request merged**